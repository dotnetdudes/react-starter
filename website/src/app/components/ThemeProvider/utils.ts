import { ThemeType } from './slice/types';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeType) || null
    : null;
}
