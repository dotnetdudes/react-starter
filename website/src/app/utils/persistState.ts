// @flow
// https://egghead.io/lessons/javascript-redux-persisting-the-state-to-the-local-storage
// https://medium.com/@jrcreencia/persisting-redux-state-to-local-storage-f81eb0b90e7e
import type { RootState } from 'types';

export const loadState = () => {
  try {
    const serializedState = sessionStorage.getItem('stateSramReg');
    if (serializedState === null) {
      return undefined;
    }
    // $FlowIgnore
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

export const saveState = (state: RootState) => {
  try {
    const serializedState = JSON.stringify(state);
    sessionStorage.setItem('stateSramReg', serializedState);
  } catch (err) {
    // die
  }
};
