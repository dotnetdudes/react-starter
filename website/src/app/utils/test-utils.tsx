import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureAppStore } from 'store/configureStore';
import { ThemeProvider, createTheme } from '@mui/material/styles';

const store = configureAppStore();

const MockTheme = ({ children }: any) => {
  const theme = createTheme({});
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <Provider store={store}>
      <MockTheme>
        <BrowserRouter>{children}</BrowserRouter>
      </MockTheme>
    </Provider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
