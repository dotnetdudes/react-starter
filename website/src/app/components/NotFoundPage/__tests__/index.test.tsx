import { NotFoundPage } from '../index';
import { render } from 'app/utils/test-utils';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

jest.mock('react-helmet-async', () => {
  const React = require('react');
  const plugin = jest.requireActual('react-helmet-async');
  const mockHelmet = ({ children, ...props }) =>
    React.createElement(
      'div',
      {
        ...props,
        className: 'mock-helmet',
      },
      children,
    );
  return {
    ...plugin,
    Helmet: jest.fn().mockImplementation(mockHelmet),
  };
});

describe('<NotFoundPage  />', () => {
  it('should match snapshot', () => {
    const notFound = render(<NotFoundPage />);
    expect(notFound.container.firstChild).toMatchSnapshot();
  });

  it('should render a <div> tag', () => {
    const notFound = render(<NotFoundPage />);
    expect(notFound.container.querySelector('div')).toBeInTheDocument();
  });
});
