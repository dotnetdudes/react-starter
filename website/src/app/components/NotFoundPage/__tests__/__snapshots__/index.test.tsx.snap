// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage  /> should match snapshot 1`] = `
<div
  class="MuiBox-root css-1xy9oj"
>
  <div>
    <div>
      4
      <span
        aria-label="Crying Face"
        role="img"
      >
        ðŸ˜¢
      </span>
      4
    </div>
    <div>
      NotFoundPage.title
    </div>
  </div>
</div>
`;
